test/unit/org/apache/cassandra/utils/TeeDataInputPlusTest.java

```diff
@@ -0,0 +1,141 @@
+        // String
+        out.writeUTF("abc");
+        //Another string to test skip
+        out.writeUTF("garbagetoskipattheend");
+        testData = out.toByteArray();
+
+        int LIMITED_SIZE = 40;
+        DataInputBuffer reader = new DataInputBuffer(testData);
+        DataInputBuffer reader2 = new DataInputBuffer(testData);
+        DataOutputBuffer teeOut = new DataOutputBuffer();
+        DataOutputBuffer limitedTeeOut = new DataOutputBuffer();
+        TeeDataInputPlus tee = new TeeDataInputPlus(reader, teeOut);
+        TeeDataInputPlus limitedTee = new TeeDataInputPlus(reader2, limitedTeeOut, LIMITED_SIZE);
+
+        // boolean = 1byte
+        boolean bool = tee.readBoolean();
+        assertTrue(bool);
+        bool = limitedTee.readBoolean();
+        assertTrue(bool);
+        // byte = 1byte
+        byte b = tee.readByte();
+        assertEquals(b, 0x1);
+        b = limitedTee.readByte();
+        assertEquals(b, 0x1);
+        // char = 2byte
+        char c = tee.readChar();
+        assertEquals('a', c);
+        c = limitedTee.readChar();
+        assertEquals('a', c);
+        // short = 2bytes
+        short s = tee.readShort();
+        assertEquals(1, s);
+        s = limitedTee.readShort();
+        assertEquals(1, s);
+        // int = 4bytes
+        int i = tee.readInt();
+        assertEquals(1, i);
+        i = limitedTee.readInt();
+        assertEquals(1, i);
+        // long = 8bytes
+        long l = tee.readLong();
+        assertEquals(1L, l);
+        l = limitedTee.readLong();
+        assertEquals(1L, l);
+        // float = 4bytes
+        float f = tee.readFloat();
+        assertEquals(1.0f, f, 0);
+        f = limitedTee.readFloat();
+        assertEquals(1.0f, f, 0);
+        // double = 8bytes
+        double d = tee.readDouble();
+        assertEquals(1.0d, d, 0);
+        d = limitedTee.readDouble();
+        assertEquals(1.0d, d, 0);
+        long vint = tee.readVInt();
+        assertEquals(-1337L, vint);
+        vint = limitedTee.readVInt();
+        assertEquals(-1337L, vint);
+        long uvint = tee.readUnsignedVInt();
+        assertEquals(1337L, uvint);
+        uvint = limitedTee.readUnsignedVInt();
+        assertEquals(1337L, uvint);
+        // String("abc") = 2(string size) + 3 = 5 bytes
+        String str = tee.readUTF();
+        assertEquals("abc", str);
+        str = limitedTee.readUTF();
+        assertEquals("abc", str);
+        int skipped = tee.skipBytes(100);
+        assertEquals(23, skipped);
+        skipped = limitedTee.skipBytes(100);
+        assertEquals(23, skipped);
+
+        byte[] teeData = teeOut.toByteArray();
+        assertFalse(tee.isLimitReached());
+        assertTrue(Arrays.equals(testData, teeData));
+
+        byte[] limitedTeeData = limitedTeeOut.toByteArray();
+        assertTrue(limitedTee.isLimitReached());
+        assertTrue(Arrays.equals(Arrays.copyOf(testData, LIMITED_SIZE -1 ), limitedTeeData));
```

Comment:
I'm not sure I follow, do you mean the `LIMITED_SIZE-1`?  Since the limit is `LIMITED_SIZE` the buffer stopped copying at `LIMITED_SIZE-1` bytes

Reply:
Oh. 🤦 This is a formatting issue - missing a whitespace between the - and the 1. I was literally reading
`[int] [-int]` instead of `[int] - [int]`.

So this should be
```
assertTrue(Arrays.equals(Arrays.copyOf(testData, LIMITED_SIZE - 1 ), limitedTeeData));
```